util/httputil/http_util.go

```diff
@@ -67,7 +67,25 @@ func newAuthRoundTripper(auth *config.Auth, rt http.RoundTripper) http.RoundTrip
 	}
 }
 
-func AuthTransport(auth *config.Auth, transportConfig *http.Transport) (http.RoundTripper, error) {
+// Creates a new HTTP Transport with TLS and Timeouts.
+//
+// Please remember that setting long timeouts is not recommended as it can make
+// idle connections stay open for as long as 2 * timeout. This should only be
+// done in cases where you know the request is very likely going to be reused at
+// some point in the near future.
+func CreateTransport(auth *config.Auth, transportConfig *http.Transport, timeout time.Duration) (http.RoundTripper, error) {
+	// Limits the time spent establishing a TCP connection if a new one is
+	// needed.
+	transportConfig.Dial = (&net.Dialer{
```

Comment:
I'm not sure about this, if there's already a dial (dialcontext?) defined in Transport, you are overwriting it. And in the case of prometheus we use the `DefaultRoundTripper` which actually defines a dialer, with KeepAlive: https://github.com/prometheus/client_golang/blob/master/api/client.go#L29-L36

(I'm not sure why there is `Transport.Dial` and `Transport.DialContext`, and how the two of them relate).

Perhaps this should be executed only if Dial and DialContext are null?

Reply:
In general I prefer to have it set always, but I do see your point. Added it.